frontend/src/app/libs/common-components/components/input/input.component.html

```diff
@@ -0,0 +1,40 @@
+<mat-form-field 
+	appearance="outline" 
+	class="input-component"
+	[class.input-component--dark]="isDarkTheme">
+
+	<mat-label>{{label}}</mat-label>
+	<input
+		matInput 
+		class="ng-touched"
+		[formControl]="formControlName"
+		[type]="type"
+		placeholder={{placeholder}}>
+	<mat-error *ngIf="isText && textFormControl.hasError('required')">
+		Please enter a {{inputType}} 
```

Comment:
lets generate error labels dynamically, and make consumer do it.
please create `[validators]` which gets array of functions `ValidatorFn[]` https://angular.io/guide/form-validation#defining-custom-validators which compose new button state, including errors

Reply:
Also please make sure your component can be used in `ngForm` as field.